article/app/controllers/ArticleController.scala

```diff
@@ -234,15 +250,25 @@ class ArticleController extends Controller with RendersItemResponse with Logging
 }
 
 object ParseBlockId extends RegexParsers {
-  def apply(input: String): Option[String] = {
-    def withParser: Parser[Unit] = "with:" ^^ { _ => () }
-    def block: Parser[Unit] = "block-" ^^ { _ => () }
-    def id: Parser[String] = "[a-zA-Z0-9]+".r
-    def expr: Parser[String] = withParser ~> block ~> id
+
+  def withParser: Parser[Unit] = "with:" ^^ { _ => () }
+  def block: Parser[Unit] = "block-" ^^ { _ => () }
+  def id: Parser[String] = "[a-zA-Z0-9]+".r
+  def blockId = block ~> id
```

Comment:
Would it make sense to make those private if they are not supposed to be exposed outside of this object?